packages/app-root/src/app/style-registry.js

```diff
@@ -0,0 +1,27 @@
+'use client'
+
+import { useServerInsertedHTML } from 'next/navigation'
+import { ServerStyleSheet, StyleSheetManager } from 'styled-components'
+import { useState } from 'react'
+
+export default function StyledComponentsRegistry({
+  children
+}) {
+  // Only create stylesheet once with lazy initial state
+  // x-ref: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
+  const [styledComponentsStyleSheet] = useState(() => new ServerStyleSheet())
+
+  useServerInsertedHTML(() => {
+    const styles = styledComponentsStyleSheet.getStyleElement()
+    styledComponentsStyleSheet.instance.clearTag()
+    return <>{styles}</>
+  })
+
+  if (typeof window !== 'undefined') return <>{children}</>
```

Comment:
This checks if the code is running in a browser, but the file specifies `use client`, so that's confusing.

Reply:
I'm also a bit confused. I used the example from https://nextjs.org/docs/app/building-your-application/styling/css-in-js#styled-components, but don't totally understand why this line is included.

Reply:
Good to know I'm not the only one that's confused by how this works. 😄